### PR TITLE
Add the data set manipulate method which used in client.

### PR DESCRIPTION
**Pull request by Shylock-Hg**

### CODE REVIEW

File: src/common/datatypes/DataSet.h [hunk 1]
```diff
@@ -46,6 +48,56 @@ struct DataSet {
...
+        std::vector<Value> col;
+        const auto find = std::find(colNames.begin(), colNames.end(), colName);
+        if (find == colNames.end()) {
+            return col;
+        }
+        std::size_t index = std::distance(colNames.begin(), find);
+        for (const auto &row : rows) {
+            col.emplace_back(row.values[index]);
+        }
+        return col;
```
Comment (dutor) [inline src/common/datatypes/DataSet.h:69]: `reserve` by # of rows?

File: src/common/datatypes/DataSet.h [hunk 2]
```diff
@@ -46,6 +48,56 @@ struct DataSet {
...
+    iterator end() {
+        return rows.end();
+    }
+
+    const_iterator end() const {
+        return rows.end();
+    }
+
+    std::vector<std::pair<std::string, Value>> fields() const {
+        std::vector<std::pair<std::string, Value>> fs;
```
Comment (dutor) [inline src/common/datatypes/DataSet.h:92]: reserve

Comment (dutor) [reply src/common/datatypes/DataSet.h:92]: BTW, what's this used for?

Comment (Shylock-Hg) [reply src/common/datatypes/DataSet.h:92]: It's provided for client user to manipulate the data.

File: src/common/datatypes/DataSet.h (hunk 1 above)
Comment (Shylock-Hg) [reply src/common/datatypes/DataSet.h:70]: Fixed.

**Review by dutor — APPROVED** [review]
LGTM.

**Review by yixinglu — APPROVED** [review]

**Pull request merged**